control/tests/test_deliver.py

```diff
@@ -0,0 +1,169 @@
+        self.encoded_file = self.veda_id + '_' + self.encode.encode_suffix + '.' + self.encode.encode_filetype
+
+        self.DeliverInstance.run()
+        # Assertions
+        self.assertEqual(self.DeliverInstance.video_proto.val_id, 'XXXXXXXX2014-V00TES1')
+        self.assertEqual(self.DeliverInstance.video_proto.veda_id, self.veda_id)
+        self.assertEqual(self.DeliverInstance.video_proto.duration, 10.09)
+        self.assertEqual(self.DeliverInstance.video_proto.s3_filename, None)
+
+        self.assertEqual(self.DeliverInstance.encode_query, self.encode)
+        self.assertEqual(self.DeliverInstance.encoded_file, '/'.join((self.veda_id, self.veda_id + '.m3u8')))
+        self.assertEqual(self.DeliverInstance.status, 'Complete')
+        self.assertEqual(
+            self.DeliverInstance.endpoint_url,
+            '/'.join((
+                CONFIG_DATA['edx_cloudfront_prefix'],
+                self.veda_id,
+                self.veda_id + '.m3u8'
+            ))
+        )
+
+    def test_intake(self):
+        """
+        Framework for intake testing
+        """
+        self.DeliverInstance._INFORM_INTAKE()
+        self.encoded_file = self.veda_id + '_' + self.encode.encode_suffix + '.' + self.encode.encode_filetype
+        self.assertEqual(self.DeliverInstance.encoded_file, self.encoded_file)
+        self.assertEqual(
+            self.DeliverInstance.hotstore_url,
+            '/'.join((
+                'https:/',
+                's3.amazonaws.com',
+                CONFIG_DATA['veda_deliverable_bucket'],
+                self.encoded_file
+            ))
+        )
+
+    @unittest.skip('Skipping this test due to unavailability of ffprobe req')
+    def test_validate(self):
+        """
+        Simple test of validation call from deliver, not a full test of validation function
+        """
+        self.DeliverInstance.node_work_directory = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            'test_files'
+        )
+        self.DeliverInstance.encoded_file = 'OVTESTFILE_01.mp4'
+        self.assertTrue(self.DeliverInstance._VALIDATE())
+
+    def test_determine_status(self):
+        """
+        Test Video Status determiner
+        """
+        self.DeliverInstance.video_query = self.video
+
+        self.assertEqual(self.DeliverInstance._DETERMINE_STATUS(), 'Progress')
+        self.url = URL.objects.create(
+            encode_profile=self.encode,
+            videoID=self.video,
+            encode_url='Test_URL'
+        )
+        self.assertEqual(self.DeliverInstance._DETERMINE_STATUS(), 'Complete')
+
+    def test_validate_url(self):
+        """
+        Test URL Validator
+        """
+        self.assertFalse(self.DeliverInstance._VALIDATE_URL())
+        self.DeliverInstance.endpoint_url = 'https://edx.org'
+        self.assertTrue(self.DeliverInstance._VALIDATE_URL())
+
+    @patch('control.veda_val.VALAPICall._AUTH', PropertyMock(return_value=lambda: CONFIG_DATA))
```

Comment:
this `patch` is something that should be moved to class level, let's do that.

Reply:
👍